pkg/apis/apps/v1beta1/foundationdbcluster_types_test.go

```diff
@@ -547,3 +547,1659 @@ func TestGettingSidecarVersion(t *testing.T) {
 	cluster.Spec.SidecarVersion = 2
 	g.Expect(cluster.GetFullSidecarVersion()).To(gomega.Equal("6.1.8-2"))
 }
+
+func TestGetNextConfigurationChangeWithSimpleChange(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	currentConfig := DatabaseConfiguration{
+		RedundancyMode: "double",
+	}
+	finalConfig := DatabaseConfiguration{
+		RedundancyMode: "triple",
+	}
+	nextConfig := currentConfig.GetNextConfigurationChange(finalConfig)
+	g.Expect(nextConfig).To(gomega.Equal(DatabaseConfiguration{
+		RedundancyMode: "triple",
+	}))
+}
+
+func TestGetNextConfigurationChangeWhenEnablingFearlessDR(t *testing.T) {
```

Comment:
If a recovery doesn't occur after every database configuration change, then we're probably safe to do extra steps if it makes the flow simpler

Reply:
I've changed the breakdown so it applies the changes to the satellite configuration earlier in the process. Does that address your concern, or are there other places where we should be condensing steps?

Reply:
There's still a situation when you're adding two new regions (when going from no region config to fearless) in which you are performing two steps. It's safe to add all of the regions at once since you aren't setting usable regions to 2 until after all the regions have been added.